app/actions/local/team.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Model} from '@nozbe/watermelondb';
+
+import {queryMyTeamById} from '@app/queries/servers/team';
+import DatabaseManager from '@database/manager';
+
+import type TeamModel from '@typings/database/models/servers/team';
+import type TeamMembershipModel from '@typings/database/models/servers/team_membership';
+
+export const localRemoveUserFromTeam = async (serverUrl: string, teamId: string, userId: string) => {
+    const serverDatabase = DatabaseManager.serverDatabases[serverUrl];
+    if (!serverDatabase) {
+        return;
+    }
+
+    const {operator, database} = serverDatabase;
+    const myTeam = await queryMyTeamById(database, teamId);
+    const models: Model[] = [];
+    if (myTeam) {
+        const team = await myTeam.team.fetch() as TeamModel;
+        const members: TeamMembershipModel[] = await team.members.fetch();
+        const member = members.find((m) => m.userId === userId);
+
+        myTeam.prepareDestroyPermanently();
+        models.push(myTeam);
+        if (member) {
+            member.prepareDestroyPermanently();
+            models.push(member);
+        }
+
+        if (models.length) {
+            await operator.batchRecords(models);
+        }
```

Comment:
Removing everything attached to the team is a good idea, but deleting the team itself, I am not 100% sure. In some places (team sidebar) we need to be aware if the team is joinable. Not sure if we should refetch the teams after removing it, to make sure it is synchronized (and if I have access to it (i.e. is not a private team)).

Reply:
I rather remove everything and perhaps only keep the "Team" instead of just removing "My Team" as it will by not keeping stale and unaccessible data in the databases

Reply:
In fact on app entry, when you are no longer part of a team, it will then remove the entire thing